test/Numerics/DGMethods/compressible_navier_stokes_equations/shared_source/grids.jl

```diff
@@ -190,3 +190,116 @@ function DiscretizedDomain(
         grid,
     )
 end
+
+## SphericalShellDomain
+
+"""
+    cubedshellwarp(a, b, c, R = max(abs(a), abs(b), abs(c)))
+
+Given points `(a, b, c)` on the surface of a cube, warp the points out to a
+spherical shell of radius `R` based on the equiangular gnomonic grid proposed by
+[Ronchi1996](@cite)
+"""
+function cubedshellwarp(a, b, c, R = max(abs(a), abs(b), abs(c)))
+
+    function f(sR, ξ, η)
+        X, Y = tan(π * ξ / 4), tan(π * η / 4)
+        x1 = sR / sqrt(X^2 + Y^2 + 1)
+        x2, x3 = X * x1, Y * x1
+        x1, x2, x3
+    end
+
+    fdim = argmax(abs.((a, b, c)))
+    if fdim == 1 && a < 0
+        # (-R, *, *) : Face I from Ronchi, Iacono, Paolucci (1996)
+        x1, x2, x3 = f(-R, b / a, c / a)
+    elseif fdim == 2 && b < 0
+        # ( *,-R, *) : Face II from Ronchi, Iacono, Paolucci (1996)
+        x2, x1, x3 = f(-R, a / b, c / b)
+    elseif fdim == 1 && a > 0
+        # ( R, *, *) : Face III from Ronchi, Iacono, Paolucci (1996)
+        x1, x2, x3 = f(R, b / a, c / a)
+    elseif fdim == 2 && b > 0
+        # ( *, R, *) : Face IV from Ronchi, Iacono, Paolucci (1996)
+        x2, x1, x3 = f(R, a / b, c / b)
+    elseif fdim == 3 && c > 0
+        # ( *, *, R) : Face V from Ronchi, Iacono, Paolucci (1996)
+        x3, x2, x1 = f(R, b / c, a / c)
+    elseif fdim == 3 && c < 0
+        # ( *, *,-R) : Face VI from Ronchi, Iacono, Paolucci (1996)
+        x3, x2, x1 = f(-R, b / c, a / c)
+    else
+        error("invalid case for cubedshellwarp: $a, $b, $c")
+    end
+
+    return x1, x2, x3
+end
```

Comment:
This exact code snippet already exists exists [here](https://github.com/CliMA/ClimateMachine.jl/blob/master/test/Numerics/Mesh/filter.jl#L401-L434). Can we somehow share this instead?

Reply:
> This exact code snippet already exists exists [here](https://github.com/CliMA/ClimateMachine.jl/blob/master/test/Numerics/Mesh/filter.jl#L401-L434). Can we somehow share this instead?

We should definitely share it. The code snippet is already in https://github.com/CliMA/ClimateMachine.jl/blob/master/src/Numerics/Mesh/Topologies.jl#L2022. The keyword arguments made it so that the function could not be called in a way that repeated previous behavior. I'm open to suggestions on the best way of moving forward on that

Reply:
@sandreza , @blallen In PR #2060 I had renamed `cubedshellwarp` as  `cubed_sphere_warp` and so on for the unwarp version and added two specialized case wrappers for the multiple dispatch (I preferred using `snake_case` because it is more readable/accessible). I reckon that I didn't update the `cubedshelltopowarp` function because it was not clear to us in the atmos group how/where that was used. It would be nice if now that you are touching this, the interface and function names could be unified a bit. Thanks

Reply:
oh! is that where it went? in that case I can probably remove this commit https://github.com/CliMA/ClimateMachine.jl/pull/2171/commits/6d4c7eaba50683d11eabfda81cd9057d4536f3b4 and just call `cubed_sphere_warp`

Reply:
it worked! yay fewer changes!

Reply:
Thanks @blallen . So, the `cubedshelltopowarp` and `cubedshelltopounwarp` are not needed at all then? Should they be renamed at least to follow the current convention of the other ones?

Reply:
@blallen not sure if you saw this. I can rename them in a separate quick PR

Reply:
Added #2181

Reply:
sorry for the delay; gmail has decided everything from github is spam :'(

I think the `topowarp` versions are different functions? so #2181 is probably the way to go